codec-http/src/main/java/io/netty/handler/codec/http/websocketx/extensions/compression/DeflateFrameClientExtensionHandshaker.java

```diff
@@ -61,16 +110,19 @@ public DeflateFrameClientExtensionHandshaker(int compressionLevel, boolean useWe
      *            Compression level between 0 and 9 (default is 6).
      * @param extensionFilterProvider
      *            provides client extension filters for per frame deflate encoder and decoder.
+     * @param maxAllocation
+     *            Maximum size of the decompression buffer. Must be &gt;= 0. If zero, maximum size is not limited.
      */
     public DeflateFrameClientExtensionHandshaker(int compressionLevel, boolean useWebkitExtensionName,
-            WebSocketExtensionFilterProvider extensionFilterProvider) {
+            WebSocketExtensionFilterProvider extensionFilterProvider, int maxAllocation) {
         if (compressionLevel < 0 || compressionLevel > 9) {
             throw new IllegalArgumentException(
                     "compressionLevel: " + compressionLevel + " (expected: 0-9)");
         }
         this.compressionLevel = compressionLevel;
         this.useWebkitExtensionName = useWebkitExtensionName;
         this.extensionFilterProvider = checkNotNull(extensionFilterProvider, "extensionFilterProvider");
+        this.maxAllocation = maxAllocation;
```

Comment:
Should we check that this is >= 0 ?. There are utilities in `ObjectUtils` for this. This is true for all constructors

Reply:
Added to all public constructors